lib/path.js

```diff
@@ -1,7 +1,14 @@
 'use strict';
 
+const util = require('util');
 const isWindows = process.platform === 'win32';
 
+function assertPath(path) {
+  if (typeof path !== 'string')
+    throw new TypeError('Path must be a string. Received ' +
+                        util.inspect(path));
+}
+
```

Comment:
I'd rather not have this be over-engineered, we're just checking for argument types - anyways, as is, you could see which argument is bad via the stack trace.